packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -257,7 +258,7 @@ export async function getBlock<ReturnFormat extends DataFormat>(
 			hydrated,
 		);
 	}
-	return format(blockSchema, response as unknown as Block, returnFormat);
```

Comment:
I believe you would want to edit `blockSchema` and add the new properties there, not editing blockHeaderSchema and just using blockSchema

Reply:
> I believe you would want to edit `blockSchema` and add the new properties there, not editing blockHeaderSchema and just using blockSchema

Ok, yeah I'd noticed the `withdrawalsRoot` property only appears in `blockHeaderSchema`, under `schemas.ts`. When I call getBlock() on web3.js v1.9, it includes the `withdrawalsRoot` property. This is why I'd assumed `getBlock()` needed `blockHeaderSchema` in v4 instead of `blockSchema`.

Should `blockSchema` & `blockHeaderSchema` have identical properties then?

Reply:
Yea its a bit confusing, It seems like `blockSchema` needs to also have updated properties of `withdrawals` and `withdrawalsRoot`. `blockHeaderSchema` is only being used for [subscriptions](https://github.com/web3/web3.js/blob/526c6f518f6a23438907b119aadbc868b06fbcd6/packages/web3-eth/src/web3_subscriptions.ts#L121). We may need to update the blockHeaderSchema as well, if you want to check and confirm result of a subscription the result you can also edit blockheaderSchema, otherwise you can focus on just updating blockSchema

Reply:
Ok, I've updated both blockSchema & blockHeaderSchema in `schemas.ts`, with blockHeaderSchema retaining the below author & excessDataGas properties while missing from blockSchema. Please let me know if blockSchema should also have these.

blockSchema & blockHeaderSchema also have different nested arrays for the `transactions` property, and were left as originally found.

```
author: {
   format: 'bytes32',
},
excessDataGas: {
   format: 'uint',
},
```
The BlockOutput & BlockHeaderOutput interfaces in `eth_types.ts` were also updated accordingly.

Under eth_types.ts, I'd noticed totalDifficulty under BlockOutput is formatted as:
`readonly totalDifficulty?: bigint | number;`

Whereas on BlockHeaderOutput, totalDifficulty is formatted as:
`readonly totalDifficulty?: Numbers; `

I'm unsure if it should be one or the other, but I've left as-is. Also the extra properties: author & excessDataGas were not added to the BlockOutput interface, as they are not present in blockSchema.

Let me know if those author & excessDataGase properites should also be present across blockSchema/blockHeaderSchema & BlockOutput/BlockHeaderOutput.

`yarn.lock` also had its changes reverted, as well as `rpc_method_wrappers.ts`.

Reply:
Adding a note here so I don't forget after sleep. Looks like more parameters are missing from v4.6 yet present in v1.9, outside the scope of EIP-4844 changes, specifically across other `eth_*` functions. I'll do a much proper rundown thru the next couple of days, likely into the weekend.

For ex: the `id` parameter is missing in the `logs` array, from eth_getTransactionReceipt. Example below from an ETH mainnet request:

web3 v4.6:
`{"address":"0xb584d4be1a5470ca1a8778e9b86c81e165204599","blockHash":"0x47c58ebbaaff9a9a9ee60d74029224df7cac63b5c09972d4306e7500d8078b55","blockNumber":"19529894","data":"0x0000000000000000000000000000000000000000000000000000022ecb25c000","logIndex":"469","removed":false,"topics":["0x90890809c654f11d6e72a28fa60149770a0d11ec6c92319d6ceb2bb0a4ea1a15","0x0000000000000000000000000a25e45792fef9fb600a81f0d7ac4ab9e893fc71","0x0000000000000000000000000000000000000000000000000000000000000064"],"transactionHash":"0xe2bad713807549fe3bb0bf0a69dce0ddc634c8dbec655660ea038279ef5bdb91","transactionIndex":"170"}`

web3 v1.9:
`{"address":"0xb584D4bE1A5470CA1a8778E9B86c81e165204599","blockHash":"0x47c58ebbaaff9a9a9ee60d74029224df7cac63b5c09972d4306e7500d8078b55","blockNumber":19529894,"data":"0x0000000000000000000000000000000000000000000000000000022ecb25c000","id":"log_ff3f9ddd","logIndex":469,"removed":false,"topics":["0x90890809c654f11d6e72a28fa60149770a0d11ec6c92319d6ceb2bb0a4ea1a15","0x0000000000000000000000000a25e45792fef9fb600a81f0d7ac4ab9e893fc71","0x0000000000000000000000000000000000000000000000000000000000000064"],"transactionHash":"0xe2bad713807549fe3bb0bf0a69dce0ddc634c8dbec655660ea038279ef5bdb91","transactionIndex":170}`